keps/sig-node/5067-pod-generation/README.md

```diff
@@ -0,0 +1,1095 @@
+behavior of the `metadata.generation` field in all other objects. 
+
+#### Kubelet and observedGeneration
+
+When the Kubelet updates the pod status as part of the [pod sync loop](https://github.com/kubernetes/kubernetes/blob/45d0fddaf1f24f7b559eb936308ce2aeb9871850/pkg/kubelet/pod_workers.go#L1214), 
+it will set the `status.observedGeneration` in the pod to reflect the pod `metadata.generation` corresponding 
+to the snapshot of the pod currently being synced. That means if the 
+pod spec gets updated concurrently while the kubelet is performing a pod sync loop 
+on a previous update, the `status.observedGeneration` will be behind `metadata.generation`. 
+
+Outside of the pod sync loop, another place where the kubelet sets the pod status is
+when a [pod is rejected (during `HandlePodAdditions`)](https://github.com/kubernetes/kubernetes/blob/8770bd58d04555303a3a15b30c245a58723d0f4a/pkg/kubelet/kubelet.go#L2335). This
+code will also be modified to populate `status.observedGeneration` to express which
+`metadata.generation` was rejected. In this case, the kubelet will not be updating the pod
+through the sync loop (due to the rejection).
+
+The only other place where the pod status is updated is the readiness and probe
+updates, but we will leave `status.observedGeneration` unchanged here as the probe that
+updated the status would be the one synced in the last pod sync loop.
+
+##### Other writers of pod status
+
+There are other writers of pod status besides the Kubelet including the scheduler
+and the node lifecycle controller. These should also populate `status.observedGeneration` whenever 
+they make a status update (excluding any updates to pod conditions, which
+have their own dedicated `observedGeneration` field). Once the pod is bound to a node, 
+however, the expectation is that only the kubelet will be writing to `status.observedGeneration`. 
+
+The scheduler and node lifecycle controller also write pod conditions. Whenever 
+they set a pod condition, they should just populate the `condition.observedGeneration` field 
+with the relevant generation.
+
+#### Client requests to update observedGeneration
+
+During status update, if the incoming update clears `status.observedGeneration` back
+to 0, the API server will preserve the previously existing value. All other updates to `status.observedGeneration` will be permitted by the API validation, including
+regressions back to decreasing values.
+
+#### Mirror pods
+
+We will consider populating `metadata.generation` and `status.observedGeneration` out of scope for mirror 
+pods. This means:
+
+* The API server never sets `metadata.generation` for mirror pods.
+* If there is `metadata.generation` or `status.observedGeneration` set in the incoming config for a static pod, we ignore and drop them on the API side.
+* Kubelet never sets `status.observedGeneration` for mirror pods (it will need to explicitly exclude them).
```

Comment:
I know this is out of scope, but it'd be good to include some reasons here so we don't rehash the same thing, e.g., any update to a static pod will be considered a new pod by kubelet since kubelet computes the internal pod UID using the pod spec. This defeats the purpose of generation, which is for tracking versions. To fully support generation for mirror pods, more changes to kubelet's logic will be expected.

Reply:
thanks! Added the reasons to the kep